src/Microsoft.AspNet.Identity/SignInManager.cs

```diff
@@ -403,6 +411,59 @@ public AuthenticationProperties ConfigureExternalAuthenticationProperties(string
             return null;
         }
 
+        /// <summary>
+        ///     Log boolean result for user and return result
+        /// </summary>
+        /// <param name="result"></param>
+        /// <param name="user"></param>
+        /// <param name="methodName"></param>
+        /// <returns></returns>
+        protected async virtual Task<bool> LogResultAsync(bool result, TUser user, [System.Runtime.CompilerServices.CallerMemberName] string methodName = "")
+        {
+            Logger.WriteInformation(Resources.FormatLoggingSigninStatus(methodName,
+                await UserManager.GetUserIdAsync(user), result));
+
+            return result;
+        }
+
+        /// <summary>
+        ///     Log SignInStatus for user and return SignInStatus
+        /// </summary>
+        /// <param name="status"></param>
+        /// <param name="user"></param>
+        /// <param name="methodName"></param>
+        /// <returns></returns>
+        protected async virtual Task<SignInResult> LogResultAsync(SignInResult status, TUser user, [System.Runtime.CompilerServices.CallerMemberName] string methodName = "")
+        {
+            if (status.IsLockedOut)
```

Comment:
Would it be better to move some of this functionality to a Log method in SignInResult similar to how we do it with IdentityResult?